hash/hash.go

```diff
@@ -0,0 +1,101 @@
+// Package hash provides utility functions to process complex data types, like
+// data streams, files, or sequences of structs of different types.
+package hash
+
+import (
+	"bytes"
+	"errors"
+	h "hash"
+	"io"
+	"os"
+
+	"encoding"
+	"reflect"
+)
+
+var defaultChunkSize = 1024
```

Comment:
Done, except for the renaming of `Args`. Maybe `Arguments` to be in line with `Stream` and `File`?